### DISCUSSION TITLE
How can I use like query in $match ?

### DISCUSSION THREAD

**Discussion by vishnukumarps** [loopbackio/loopback-next#8125] [Q&A] [answered]
` const notificationCollection = (this.notificationRepository.dataSource.connector as any).collection("Notification");

   var msg="test msg";
    return await notificationCollection
      .aggregate([
        {$match: {message: msg}}, //  I want to add a like query here
        ]).get()
        `
      How can I add a like query inside $match?
      
   I know how to get it done by below method. I am looking for a solution in aggregate function
       await this.notificationRepo.find({
      where: {
        message: {
          like: ".*" + msg + ".*",
        }
      }
    });

**Comment by achrinza** [comment, marked as answer]
Using the underlying MongoDB driver bypasses LoopBack's filter logic.

Behind the scenes, `like` uses MongoDB's `$regex`:

https://github.com/loopbackio/loopback-connector-mongodb/blob/e985dd7a194ae56fac833adec8ba5efdec5c6c89/lib/mongodb.js#L1042-L1046

**Reply by vishnukumarps** [reply]
thanks for the reply
